Source/AssetRipper.Processing/AnimationClips/CustomCurveResolver.cs

```diff
@@ -124,398 +130,195 @@ public string ToAttributeName(BindingCustomType type, uint attribute, string pat
-							case 10:
-								return $"m_TranslationOffsets.Array.data[{attribute >> 8}].z";
-							case 11:
-								return $"m_RotationOffsets.Array.data[{attribute >> 8}].x";
-							case 12:
-								return $"m_RotationOffsets.Array.data[{attribute >> 8}].y";
-							case 13:
-								return $"m_RotationOffsets.Array.data[{attribute >> 8}].z";
-							case 14:
-								return $"m_Sources.Array.data[{attribute >> 8}].sourceTransform";
-							case 15:
-								return $"m_Sources.Array.data[{attribute >> 8}].weight";
-						}
+						return property switch
+						{
+							0 => "m_Weight",
+							1 => "m_AffectTranslationX",
+							2 => "m_AffectTranslationY",
+							3 => "m_AffectTranslationZ",
+							4 => "m_AffectRotationX",
+							5 => "m_AffectRotationY",
+							6 => "m_AffectRotationZ",
+							7 => "m_Active",
+							8 => $"m_TranslationOffsets.Array.data[{attribute >> 8}].x",
+							9 => $"m_TranslationOffsets.Array.data[{attribute >> 8}].y",
+							10 => $"m_TranslationOffsets.Array.data[{attribute >> 8}].z",
+							11 => $"m_RotationOffsets.Array.data[{attribute >> 8}].x",
+							12 => $"m_RotationOffsets.Array.data[{attribute >> 8}].y",
+							13 => $"m_RotationOffsets.Array.data[{attribute >> 8}].z",
+							14 => $"m_Sources.Array.data[{attribute >> 8}].sourceTransform",
+							15 => $"m_Sources.Array.data[{attribute >> 8}].weight",
+							_ => throw new ArgumentException($"Unknown attribute {attribute} for {type}")
+						};
 					}
-					throw new ArgumentException($"Unknown attribute {attribute} for {type}");
 
-#warning TODO:
 				case BindingCustomType.LookAtConstraint:
 					{
 						uint property = attribute & 0xF;
-						switch (property)
-						{
-							case 0:
-								return "m_Weight";
-							case 1:
-								return "m_Active";
-							case 2:
-								return "m_WorldUpObject";
-							case 3:
-								return $"m_Sources.Array.data[{attribute >> 8}].sourceTransform";
-							case 4:
-								return $"m_Sources.Array.data[{attribute >> 8}].weight";
-							case 5:
-								return "m_Roll";
-						}
+						return property switch
+						{
+							0 => "m_Weight",
+							1 => "m_Active",
+							2 => "m_WorldUpObject",
+							3 => $"m_Sources.Array.data[{attribute >> 8}].sourceTransform",
+							4 => $"m_Sources.Array.data[{attribute >> 8}].weight",
+							5 => "m_Roll",
+							_ => throw new ArgumentException($"Unknown attribute {attribute} for {type}")
+						};
 					}
-					throw new ArgumentException($"Unknown attribute {attribute} for {type}");
 
 				case BindingCustomType.Camera:
 					{
-						if (attribute == Crc32Algorithm.HashAscii("field of view"))
-						{
-							return "field of view";
-						}
-						if (attribute == Crc32Algorithm.HashAscii("m_FocalLength"))
+						if (Camera.TryGetPath(attribute, out string? foundPath))
 						{
-							return "m_FocalLength";
+							return foundPath;
 						}
 					}
 					throw new ArgumentException($"Unknown attribute {attribute} for {type}");
```

Comment:
The above cases came from uTinyRipper. The 4 cases below were added later when I discovered that the enum had been expanded. They could also benefit from `TryGetPath` or a reverse crc.